src/pip/_internal/operations/install/wheel.py

```diff
@@ -552,11 +558,41 @@ def files_to_process(
                     cast('RecordPath', srcfile), destfile, src_disk_path
                 )
 
-    files = files_to_process(
+    def all_paths():
+        # type: () -> Iterable[RecordPath]
+        for dir, _subdirs, files in os.walk(
+            ensure_text(source, encoding="utf-8")
+        ):
+            basedir = dir[len(source):].lstrip(os.path.sep)
+            for f in files:
+                path = os.path.join(basedir, f).replace(os.path.sep, "/")
+                yield cast("RecordPath", path)
+
+    def root_scheme_file_maker(source, dest):
+        # type: (text_type, text_type) -> Callable[[RecordPath], File]
+        def make_root_scheme_file(record_path):
+            # type: (RecordPath) -> File
+            normed_path = os.path.normpath(record_path)
+            source_disk_path = os.path.join(source, normed_path)
+            dest_path = os.path.join(dest, normed_path)
+            return DiskFile(record_path, dest_path, source_disk_path)
+
+        return make_root_scheme_file
+
+    def is_data_scheme_path(path):
+        # type: (RecordPath) -> bool
+        return path.split("/", 1)[0].endswith(".data")
+
+    paths = all_paths()
+    root_scheme_paths, _data_scheme_paths = partition(
+        is_data_scheme_path, paths
+    )
+
+    make_root_scheme_file = root_scheme_file_maker(
```

Comment:
I don't have a strong opinion either way. I'll leave this as-is for now, and if/when we start moving some of these functions out I think we'll have a better idea about what's easier to understand (since we'd have to change them to align, too).